packages/react-ui/src/widgets/BarWidgetUI.js

```diff
@@ -0,0 +1,437 @@
+      )}
+      {!!options && (
+        <ReactEcharts
+          option={options}
+          lazyUpdate={true}
+          onEvents={onEvents}
+          style={{ height }}
+        />
+      )}
+    </div>
+  );
+}
+
+BarWidgetUI.defaultProps = {
+  tooltip: true,
+  tooltipFormatter: defaultTooltipFormatter,
+  yAxisData: [],
+  series: [],
+  xAxisFormatter: (v) => v,
+  yAxisFormatter: (v) => v,
+  selectedBars: [],
+  labels: {},
+  onSelectedBarsChange: null,
+  animation: true,
+  stacked: true
+};
+
+BarWidgetUI.propTypes = {
+  yAxisData: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.number)),
+    PropTypes.arrayOf(PropTypes.number)
+  ]).isRequired,
+  xAxisData: PropTypes.arrayOf(PropTypes.string).isRequired,
+  series: PropTypes.arrayOf(PropTypes.string),
+  colors: PropTypes.arrayOf(PropTypes.string),
+  stacked: PropTypes.bool,
+  labels: PropTypes.object,
+  tooltip: PropTypes.bool,
+  tooltipFormatter: PropTypes.func,
+  xAxisFormatter: PropTypes.func,
+  yAxisFormatter: PropTypes.func,
+  selectedBars: PropTypes.oneOfType([
+    PropTypes.arrayOf(PropTypes.arrayOf(PropTypes.number)),
+    PropTypes.arrayOf(PropTypes.number)
+  ]),
+  onSelectedBarsChange: PropTypes.func,
+  height: PropTypes.number,
+  animation: PropTypes.bool
+};
+
+export default BarWidgetUI;
+
+// Aux
+function defaultTooltipFormatter(params, xAxisFormatter, yAxisFormatter) {
+  if (!params || !params?.length) {
+    return null;
+  }
+
+  if (Array.isArray(params) && params.length) {
+    let message = '';
+    message += `${processFormatterRes(xAxisFormatter(params[0].axisValueLabel))}`;
+    message += params
+      .map(({ seriesName, value, data, marker }) => {
+        const formattedSeriesName = seriesName ? seriesName + ': ' : '';
+        const formattedValue = processFormatterRes(yAxisFormatter(value));
+        const item = `<div style="margin-left: 8px; display: inline">
+        ${formattedSeriesName}${formattedValue}${data.unit || ''}
+        </div>`;
+        return `<div style="margin-top: 4px">${marker}${item}</div>`;
+      })
+      .join(' ');
+    return message;
+  }
+}
+
+function useProcessedProps({
+  labels,
+  height,
+  selectedBars: _selectedBars,
+  yAxisData: _yAxisData,
+  colors: _colors,
+  series: _series,
+  ...props
+}) {
+  const theme = useTheme();
+
+  // Format series with labels, we don't need original series labels in the widget
+  const series = useMemo(
+    () => _series.map((name) => labels[name] || name),
+    [_series, labels]
+  );
+
+  // Use yAxisData always as a two-dimensions array
+  const yAxisData = useMemo(
+    () => (Array.isArray(_yAxisData[0]) ? _yAxisData : [_yAxisData]),
+    [_yAxisData]
+  );
+
+  // Colors
+  const colors = useMemo(() => {
+    // Use colors props if exists
+    if (_colors) return _colors;
+
+    // Use secondary.main, if yAxisData is flat or series has a unique element
+    if (yAxisData.length <= 1 || (series.lenght && series.length === 1))
```

Comment:
Typo: `series.lenght` instead of `length`.
But actually you can drop it and just test for `series.length === 1`.

Reply:
Agree. Idk why I did that.